warn-stealers.lua

```diff
@@ -0,0 +1,107 @@
+-- Script to warn when creatures that may steal food become visible
+--@ enable = true
+--[====[
+warn-stealers
+=============
+Will make a zoomable announcement whenever a creature that can eat food, guzzle drinks, or steal items enters the map and moves into a revealed location.
+Takes ``start`` or ``stop`` as parameters.
+]====]
+
+local persistTable = require("persist-table")
+local eventful = require("plugins.eventful")
+local repeatUtil = require("repeat-util")
+
+local eventfulKey = "warn-stealers"
+local numTicksBetweenChecks = 100
+
+function gamemodeCheck()
+    if df.global.gamemode ~= df.game_mode.DWARF then
+        cache = nil
+        print("warn-stealers must be run in fort mode")
+        disable()
+        return false
+    end
+    return true
+end
+
+cache = {}
+
+local races = df.global.world.raws.creatures.all
+
+function addToCacheIfStealerAndHidden(unitId)
+    if not gamemodeCheck() then
+        return
+    end
+    local unit = df.unit.find(unitId)
+    if not dfhack.units.isHidden(unit) then
+        return
+    end
+    local casteFlags = races[unit.race].caste[unit.caste].flags
+    if casteFlags.CURIOUS_BEAST_EATER or casteFlags.CURIOUS_BEAST_GUZZLER or casteFlags.CURIOUS_BEAST_ITEM then
+        cache[unit] = true
+    end
+end
+
+function announce(unit)
+    local caste = races[unit.race].caste[unit.caste]
+    local casteFlags = caste.flags
+    local desires = {}
+    if casteFlags.CURIOUS_BEAST_EATER then
+        table.insert(desires, "eat food")
+    end
+    if casteFlags.CURIOUS_BEAST_GUZZLER then
+        table.insert(desires, "guzzle drinks")
+    end
+    if casteFlags.CURIOUS_BEAST_ITEM then
+        table.insert(desires, "steal items")
+    end
+    local str = table.concat(desires, " + ")
```

Comment:
Since the rest of the message is a proper sentence, maybe " and " would be a better joiner than " + "